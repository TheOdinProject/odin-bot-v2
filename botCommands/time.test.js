const command = require('./time');
const { generateMentions } = require('./mockData');

describe('?time', () => {
  describe('regex', () => {
    it.each([
      ['?time'],
      [' ?time'],
      ['?time @odin-bot'],
      ['@odin-bot ?time'],
    ])('correct strings trigger the callback', (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['?tim'],
      ['tie'],
      ['?ti'],
      ['?times'],
      ['```function("?time", () => {}```'],
      ['?timea'],
      [''],
      [' '],
      [' ?'],
      ['@odin-bot ? time'],
      ['?tim&e'],
      ['?tim^e'],
      ['?time!'],
      ['@odin-bot? time'],
      ['https:??time.com'],
    ])("'%s' does not trigger the callback", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });

    it.each([
      ['Check this out! ?time'],
      ['Don\'t worry about ?time'],
      ['Hey @odin-bot, ?time'],
      ['?@odin-bot ^ ?me ?time ?tests$*'],
    ])("'%s' - command can be anywhere in the string", (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['@user?time'],
      ['it\'s about?time'],
      ['?timeisanillusion'],
      ['?time?'],
      ['?time*'],
      ['?time...'],
    ])("'%s' - command should be its own word/group - no leading or trailing characters", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });
  });

  describe('callback', () => {
    it('returns correct output', async () => {
      expect(await command.cb(generateMentions(0))).toMatchSnapshot();
      expect(await command.cb(generateMentions(1))).toMatchSnapshot();
      expect(await command.cb(generateMentions(2))).toMatchSnapshot();
      expect(await command.cb(generateMentions(3))).toMatchSnapshot();
    });
  });
});
