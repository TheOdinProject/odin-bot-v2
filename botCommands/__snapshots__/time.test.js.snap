// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`?time callback returns correct output 1`] = `"Time is an illusion. Please read this: https://discord.com/channels/505093832157691914/505093832157691916/765633002393829389"`;

exports[`?time callback returns correct output 2`] = `"Time is an illusion, <@1>. Please read this: https://discord.com/channels/505093832157691914/505093832157691916/765633002393829389"`;

exports[`?time callback returns correct output 3`] = `"Time is an illusion <@1> and <@2>. Please read this: https://discord.com/channels/505093832157691914/505093832157691916/765633002393829389"`;

exports[`?time callback returns correct output 4`] = `"Time is an illusion <@1>, <@2>, and <@3>. Please read this: https://discord.com/channels/505093832157691914/505093832157691916/765633002393829389"`;
